plugins/webapi/snapshot/plugin.go

```diff
@@ -0,0 +1,103 @@
+package snapshot
+
+import (
+	"os"
+	"sync"
+
+	"github.com/iotaledger/goshimmer/packages/ledgerstate"
+	"github.com/iotaledger/goshimmer/packages/mana"
+	"github.com/iotaledger/goshimmer/plugins/messagelayer"
+	"github.com/iotaledger/goshimmer/plugins/webapi"
+
+	"github.com/iotaledger/hive.go/identity"
+	"github.com/iotaledger/hive.go/node"
+	"github.com/labstack/echo"
+)
+
+// region Plugin ///////////////////////////////////////////////////////////////////////////////////////////////////////
+
+const (
+	snapshotFileName = "snapshot.bin"
+)
+
+var (
+	// plugin holds the singleton instance of the plugin.
+	plugin *node.Plugin
+
+	// pluginOnce is used to ensure that the plugin is a singleton.
+	once sync.Once
+)
+
+// Plugin returns the plugin as a singleton.
+func Plugin() *node.Plugin {
+	once.Do(func() {
+		plugin = node.NewPlugin("WebAPI snapshot Endpoint", node.Enabled, func(*node.Plugin) {
+			webapi.Server().GET("snapshot", DumpCurrentLedger)
+		})
+	})
+
+	return plugin
+}
+
+// endregion ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// region DumpCurrentLedger ///////////////////////////////////////////////////////////////////////////////////////////////////
+
+// DumpCurrentLedger dumps a snapshot (all unspent UTXO and all of the access mana) from now.
+func DumpCurrentLedger(c echo.Context) (err error) {
+	snapshot := messagelayer.Tangle().LedgerState.SnapshotUTXO()
+
+	aMana, err := snapshotAccessMana()
```

Comment:
First we fetch the transactions that are older, than `utxoFetchTime-minAge`. Then we get the access mana at `utxoFetchTime+Delta`. But are we sure that this access mana state corresponds to utxo state in `snapshot`? I don't think so.
We are lacking the ablility to tell the exact access mana state with respect to a certain utxo state. It is only a problem if within that small time window `Delta` some big mana moves are happening. And even then, if `Delta` is reasonably small, it doesn't matte rthat much.
Another thing I would liek to discuss is the fact that we should save the base access manas in the snapshot as well, not just the effective value.

Reply:
This is ok because we dont need to be that precise for the global snapshot tool for now. All that should happen is that for the most recent transactins they might not be part of the new network. if value messages become important in Nectar we can even inform the community that any spent outputs in the next 5 min or so might not be account for after the snapshot. 

Eventually we have to replace global snapshot by rolling snapshots anyways, which have to be treated quite different.